src/test/suite/recorded.test.ts

```diff
@@ -0,0 +1,106 @@
+import * as assert from "assert";
+import { promises as fsp } from "fs";
+import * as path from "path";
+import * as yaml from "js-yaml";
+import * as vscode from "vscode";
+import { isMatch } from "lodash";
+import { TestCaseFixture } from "../../TestCase";
+import { SymbolColor } from "../../constants";
+import { ThatMark } from "../../ThatMark";
+import NavigationMap from "../../NavigationMap";
+import * as sinon from "sinon";
+import { Clipboard } from "../../Clipboard";
+import { takeSnapshot } from "../../takeSnapshot";
+import { SerializedPosition, SerializedSelection } from "../../serializers";
+
+function deserializePosition(position: SerializedPosition) {
+  return new vscode.Position(position.line, position.character);
+}
+
+function deserializeSelection(
+  selection: SerializedSelection
+): vscode.Selection {
+  const active = deserializePosition(selection.active);
+  const anchor = deserializePosition(selection.anchor);
+  return new vscode.Selection(anchor, active);
+}
+
+suite("recorded test cases", async function () {
+  const directory = path.join(__dirname, "../../../testFixtures");
+  const files = await fsp.readdir(directory);
+
+  teardown(() => {
+    sinon.restore();
+  });
+
+  files.forEach(async (file) => {
+    test(file.split(".")[0], async function () {
+      this.timeout(100000);
+      const cursorless = vscode.extensions.getExtension("pokey.cursorless");
+
+      if (cursorless == null) {
+        throw new Error("Could not get cursorless extension");
+      }
+
+      const cursorlessApi: {
+        thatMark: ThatMark;
+        navigationMap: NavigationMap;
+      } = await cursorless.activate();
+      const buffer = await fsp.readFile(path.join(directory, file));
+      const fixture = yaml.load(buffer.toString()) as TestCaseFixture;
+
+      await vscode.commands.executeCommand("workbench.action.closeAllEditors");
+      const document = await vscode.workspace.openTextDocument({
+        language: fixture.languageId,
+        content: fixture.initialState.document,
+      });
+      const editor = await vscode.window.showTextDocument(document);
+      editor.selections =
+        fixture.initialState.selections.map(deserializeSelection);
+
+      if (fixture.initialState.thatMark) {
+        const initialThatMark = fixture.initialState.thatMark.map((mark) => ({
+          selection: deserializeSelection(mark),
+          editor,
+        }));
+        cursorlessApi.thatMark.set(initialThatMark);
+      }
+
+      if (fixture.initialState.clipboard) {
+        let mockClipboard = fixture.initialState.clipboard;
+        sinon.replace(Clipboard, "readText", async () => mockClipboard);
+        sinon.replace(Clipboard, "writeText", async (value: string) => {
+          mockClipboard = value;
+        });
+      }
+
+      // Wait for cursorless to set up decorations
+      await new Promise((resolve) => setTimeout(resolve, 300));
```

Comment:
Yeah, I haven't played with the timeout, it could probably be lower. Maybe we add a function/property to see if decorations are finished that we can check periodically?

Reply:
Let's leave it for now. They run pretty quick. If it becomes a problem we can do something fancier